plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseSharedMetastoreTest.java

```diff
@@ -128,4 +134,57 @@ public void testShowSchemas()
+        long v1EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot1);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(25, 'POLAND', 3, 'test 1')", schema), 1);
+        long snapshot2 = getLatestSnapshotId();
+        long v2EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot2);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(26, 'CHILE', 1, 'test 2')", schema), 1);
+        long snapshot3 = getLatestSnapshotId();
+        long v3EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot3);
+        long incorrectSnapshot = 2324324333L;
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, snapshot1), "SELECT * FROM nation");
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s", schema, timestampLiteral(v1EpochMillis)), "SELECT * FROM nation");
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d WHERE nationkey = 25", schema, snapshot2), "VALUES(25, 'POLAND', 3, 'test 1')");
+        assertQuery(format(
+                "SELECT * FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s WHERE nationkey = 25", schema, timestampLiteral(v2EpochMillis)),
+                "VALUES(25, 'POLAND', 3, 'test 1')");
+        assertQuery(format("SELECT count(*) FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, snapshot2), "VALUES(26)");
+        assertQuery(format(
+                "SELECT count(*) FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s", schema, timestampLiteral(v2EpochMillis)), "VALUES(26)");
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d WHERE nationkey = 26", schema, snapshot3), "VALUES(26, 'CHILE', 1, 'test 2')");
+        assertQuery(format(
+                "SELECT * FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s WHERE nationkey = 26", schema, timestampLiteral(v3EpochMillis)),
+                "VALUES(26, 'CHILE', 1, 'test 2')");
+        assertQuery(format("SELECT count(*) FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, snapshot3), "VALUES(27)");
+        assertQuery(format(
+                "SELECT count(*) FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s", schema, timestampLiteral(v3EpochMillis)), "VALUES(27)");
+        assertQueryFails(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, incorrectSnapshot), "Iceberg snapshot ID does not exists: " + incorrectSnapshot);
+        assertQueryFails(
+                format("SELECT * FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF TIMESTAMP '1970-01-01 00:00:00.001000000 Z'", schema),
+                format("\\QNo version history table \"%s\".nation at or before 1970-01-01T00:00:00.001Z", schema));
+    }
+
+    private long getLatestSnapshotId()
```

Comment:
Can we extract these methods to a utility class?
I think they are already multiple times duplicated.